app/src/main/java/uk/ac/ebi/atlas/experimentpage/cellplot/JsonCellPlotGenericController.java

```diff
@@ -0,0 +1,34 @@
+package uk.ac.ebi.atlas.experimentpage.cellplot;
+
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import uk.ac.ebi.atlas.controllers.JsonExceptionHandlingController;
+
+import static uk.ac.ebi.atlas.utils.GsonProvider.GSON;
+
+@RestController
+@RequestMapping(value="/json/cell-plots/generic/{experimentAccession}", method= RequestMethod.GET)
+public class JsonCellPlotGenericController extends JsonExceptionHandlingController {
+    private final CellPlotGenericDao cellPlotGenericDao;
+
+    public JsonCellPlotGenericController(CellPlotGenericDao cellPlotGenericDao) {
+        this.cellPlotGenericDao = cellPlotGenericDao;
+    }
+
+    @GetMapping(value = "/params/{plotType}",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public String getPlotParams(@PathVariable String plotType,
+                                @PathVariable String experimentAccession) {
+        var result = cellPlotGenericDao.getQueryParams(plotType, experimentAccession);
+
+        if (result.isEmpty()) {
+            throw new IllegalArgumentException("Unknown plot type " + plotType);
```

Comment:
I would prefer to do this validation well before, not in the controller as we knew the result at the DAO level. Do you have a reason to do this in the controller?

Reply:
I don't have any strong opinion on where to catch this exception. We are also doing a similar kind of exception catching in `JsonCellPlotController`. My reason for catching this exception here instead of DAO class was IMO it makes it easier to read from the controller class to see if the result is empty then it's not the right plot type.

If you still feel its better to have this in DAO class then I can move this exception there.

Reply:
I agree, please merge away as I'm approving.